welcomescreen.py

```diff
@@ -20,92 +20,92 @@
 
-            g.gameDisplay.fill(white)
-            if(ruleflag == 0):
-                g.gameDisplay.blit(background, (350, 0))
-                if play.get_rect(center=(530 + 60, 300 + 60)).collidepoint(mos_x, mos_y):
-                    g.gameDisplay.blit(pygame.transform.scale(
-                        play, (124, 124)), (530 - 2, 300 - 2))
-                    if(pygame.mouse.get_pressed())[0] == 1:
+            game.screen.fill(BLACK)
+            # Show Welcome screen
+            if ruleflag == 0:
+                game.screen.blit(self.welcomescreen, (350, 0))
+                if self.play_button.get_rect(
+                        center=(530 + 60, 300 + 60)).collidepoint(
+                            mos_x, mos_y):
+                    game.screen.blit(pygame.transform.scale(
+                        self.play_button, (124, 124)), (530 - 2, 300 - 2))
+                    if left_click_pressed:
+                        left_click_pressed = False
                         ruleflag = 1
                 else:
-                    g.gameDisplay.blit(play, (530, 300))
+                    game.screen.blit(self.play_button, (530, 300))
+            # Show Rules screen
             else:
-                g.gameDisplay.blit(rules, (350, 0))
-                g.gameDisplay.blit(button, (540, 400))
-                if play.get_rect(center=(530 + 60, 550 + 60)).collidepoint(mos_x, mos_y):
-                    g.gameDisplay.blit(pygame.transform.scale(
-                        play, (124, 124)), (530 - 2, 550 - 2))
-                    if(pygame.mouse.get_pressed())[0] == 1:
+                game.screen.blit(self.rulescreen, (350, 0))
+                if self.play_button.get_rect(
+                        center=(530 + 60, 550 + 60)).collidepoint(
+                            mos_x, mos_y):
+                    game.screen.blit(pygame.transform.scale(
+                        self.play_button, (124, 124)), (530 - 2, 550 - 2))
+                    if left_click_pressed:
                         return
                 else:
-                    g.gameDisplay.blit(play, (530, 550))
+                    game.screen.blit(self.play_button, (530, 550))
 
-            # left and right black background patches
-            pygame.draw.rect(g.gameDisplay, black, (0, 0, 350, 768))
-            pygame.draw.rect(g.gameDisplay, black, (840, 0, 693, 768))
             pygame.display.update()
-            clock.tick(60)
-
-            # Game crash or Close check
-            if crashed:
-                pygame.quit()
-                sys.exit()
-
-        # Just a window exception check condition
-        event1 = pygame.event.get()
-        if event1.type == pygame.QUIT:
-            crashed = True
-
-        if crashed:
-            pygame.quit()
-            sys.exit()
-
-if __name__ == "__main__":
-    g = welcomescreen()
-    g.run(gameDisplay)
+            clock.tick(FPS)
+        pygame.quit()
+        sys.exit()
```

Comment:
Please remove these, the activity itself handles exit events so no need to call these within the activity.

Reply:
Fixed